test/steps/task.go

```diff
@@ -0,0 +1,103 @@
+	s.Step(`^Complete "([^"]*)" task on service "([^"]*)" and process with name "([^"]*)" with body:$`, data.completeTaskOnServiceAndProcessWithName)
+}
+
+func (data *Data) serviceContainsTasksOfProcessWithNameAndTaskName(serviceName string, numberOfTasks int, processName, taskName string) error {
+	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
+	if err != nil {
+		return err
+	}
+
+	// We assume that one process instance is available, can be refactored if more flexibility needed
+	foundProcessInstances, err := framework.GetProcessInstances(data.Namespace, routeURI, processName)
+	if err != nil {
+		return err
+	}
+	if len(foundProcessInstances) == 0 {
+		return fmt.Errorf("no process instance found, expected one instance")
+	}
+	if len(foundProcessInstances) > 1 {
+		return fmt.Errorf("too many process instances found, expected one instance, but found %d instances", len(foundProcessInstances))
+	}
+
+	// We assume that only tasks with specified name exist, can be refactored if more flexibility needed
+	processInstanceID := foundProcessInstances[0]["id"].(string)
+	foundTasks, err := framework.GetTasks(data.Namespace, routeURI, processName, processInstanceID)
+	if err != nil {
+		return err
+	}
+	for _, foundTaskName := range foundTasks {
+		if taskName != foundTaskName {
+			return fmt.Errorf("found unexpected task name %s", foundTaskName)
+		}
+	}
+	if len(foundTasks) < numberOfTasks {
+		return fmt.Errorf("not enough tasks found, expected at least %d tasks, but found just %d tasks", numberOfTasks, len(foundTasks))
+	}
+
+	return nil
+}
+
+func (data *Data) completeTaskOnServiceAndProcessWithName(taskName, serviceName, processName string, body *gherkin.DocString) error {
+	routeURI, err := framework.WaitAndRetrieveRouteURI(data.Namespace, serviceName)
+	if err != nil {
+		return err
+	}
+
+	// We assume that one process instance is available, can be refactored if more flexibility needed
+	foundProcessInstances, err := framework.GetProcessInstances(data.Namespace, routeURI, processName)
+	if err != nil {
+		return err
+	}
+	if len(foundProcessInstances) == 0 {
+		return fmt.Errorf("no process instance found, expected one instance")
+	}
+	if len(foundProcessInstances) > 1 {
+		return fmt.Errorf("too many process instances found, expected one instance, but found %d instances", len(foundProcessInstances))
+	}
+
+	processInstanceID := foundProcessInstances[0]["id"].(string)
+	foundTasks, err := framework.GetTasks(data.Namespace, routeURI, processName, processInstanceID)
+	if err != nil {
+		return err
+	}
+
+	var taskID string
+	for foundTaskID, foundTaskName := range foundTasks {
+		if taskName == foundTaskName {
+			taskID = foundTaskID
+		}
+	}
+
+	err = framework.CompleteTask(data.Namespace, routeURI, processName, processInstanceID, taskName, taskID, body.ContentType, body.Content)
```

Comment:
what happens if no `taskID` found ?

Reply:
good point, will adjust the implementation